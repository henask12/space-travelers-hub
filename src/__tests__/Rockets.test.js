import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'; // For better DOM assertions
import RocketsList from '../components/Rockets';

// Mocking useDispatch and useSelector
jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  useSelector: (selectorFn) => selectorFn(
    { rockets: { rockets: [] } },
  ),
}));

test('renders the Empty Rockets component correctly', () => {
  const { container } = render(<RocketsList />);
  expect(container).toMatchSnapshot();
});
