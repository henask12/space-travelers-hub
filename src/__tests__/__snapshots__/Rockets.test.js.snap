// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Empty Rockets component correctly 1`] = `
<div>
  <div
    class="bg-white border border-gray-400 rounded-lg shadow-sm"
  >
    <div
      class="p-8"
    >
      <div
        class="bg-white border border-white-300 rounded-lg shadow-sm ml-8 mr-8"
      />
      <h2
        class="text-2xl font-semibold mb-4"
      >
        Rockets List
      </h2>
      <p>
        Loading...
      </p>
      <div
        class="grid gap-4 grid-cols-1"
      />
    </div>
  </div>
</div>
`;
